README.md

```diff
@@ -274,189 +285,506 @@ This is defined in libocpp/include/ocpp/v16/charge_point.hpp and takes the follo
-In EVerest authorization is handled by the Auth module and various auth token providers and validators. The OCPP module acts as both a token provider (for pre validated tokens in RemoteStartTransactions) and a token validator (using the authorize requests, or plug & charge)
-To use libocpp as a auth token validator (e.g. before starting a transaction) you can call the "authorize_id_token" function of the ChargePoint object
 
-## Integrate this library with your Charging Station Implementation for OCPP2.0.1
+In EVerest authorization is handled by the Auth module and various auth token providers and validators. The OCPP module acts as both a token provider (for pre validated tokens in RemoteStartTransactions) and a token validator (using the authorize requests, or plug & charge).
+
+To use libocpp as a auth token validator (e.g. before starting a transaction) you can call the "authorize_id_token" function of the ChargePoint object.
+
+## Integrating this library with OCPP 2.0.1
+
 TODO
 
+### Current Support
+
+#### Smart Charging
+
+Work is currently ongoing for the `K08 SmartCharging` portion of the OCPP 2.0.1 standard.
+
+##### Works in Progress
+
+All of these portions of Smart Charging are works in progress:
+
+- General Smart Charging
+  - K02 Central Smart
+  - K03 Local Smart Charging
+  - K04 Internal Load Balancing
+  - K05 Remote Start Transaction with Charging Profile
+  - K06 Offline Behavior Smart Charging During Transaction
+  - K07 Offline Behavior Smart Charging at Start of Transaction
+- External Charging Limit based Smart Charging
+  - K11 Set / Update External Charging Limit With Ongoing Transaction
+  - K12 Set / Update External Charging Limit Without Ongoing Transaction
+  - K13 Reset / Release External Charging Limit
+  - K14 External Charging Limit with Local Controller
+- ISO 15118 based Smart Charging
+  - K15 Charging with load leveling based on High Level Communication
+  - K16 Renegotiation initiated by CSMS
+  - K17 Renegotiation initiated by EV
+
+##### K01 SetChargingProfile
+
+Allows the CSMS to influence the charging power or current drawn from a specific EVSE or the
+entire Charging Station over a period of time.
+
+```mermaid
+sequenceDiagram
+    CSMS->>+ChargePoint : SetChargingProfileRequest(call)
+
+    ChargePoint->>+DeviceModel : SmartChargingCtrlrAvailable?
+    DeviceModel-->>-ChargePoint : Component
+
+    rect Thistle 
+    break SmartChargingCtrlrAvailable = false
+        ChargePoint-->>CSMS : Smart Charging NotSupported CallError
+    end
+    end
+
+    ChargePoint->>+SmartCharging : validate_and_add_profile(call.msg.Profile, call.msg.EVSE ID)
+
+    SmartCharging->>SmartCharging : validate_profile(Profile, EVSE ID)
+
+    rect Thistle 
+    break Invalid Profile
+        SmartCharging-->>ChargePoint : SetChargingProfileResponse: Rejected
+        ChargePoint-->>CSMS : SetChargingProfileResponse: Rejected
+    end
+    end
+
+    SmartCharging->>SmartCharging : add_profile(Profile, EVSE ID)
```

Comment:
When an add profile is completed the callback `signal_set_charging_profiles_callback`. It would probably be worth noting this.

Reply:
Check out the update.